sysdig/resource_sysdig_secure_cloud_auth_account.go

```diff
@@ -159,6 +159,10 @@ func resourceSysdigSecureCloudauthAccount() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			SchemaCloudProviderTenantId: {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
2 questions here :-
1. Don't we need to do a data.Get() and data.Set() for `SchemaCloudProviderTenantId` field in From* and To* methods?
2. Since this is a `required` field, what will be the value passed for `provider_tenant_id` in case of other providers such as GCP?

Reply:
This is only required for Azure, i don't think we can mark it as required for all vendors. Can we possibly validate after receiving the inputs?

Reply:
Yes if we mark it as `Required` but don't supply the field in case of other vendors, TF itself would complain. Like you mentioned, instead we can validate and enforce it after receiving the inputs. We can validate it (for Azure) while doing `data.Get()` within the `*FromResourceData()` method.

Reply:
@raghavanandan lets change this to Optional? now that you are taking care of reading this value only for Azure in L525

Reply:
Thanks!